src/batou/lib/appenv.py

```diff
@@ -20,6 +21,10 @@ def update(self):
                  "{{component.parent.env_dir}}")
         self.cmd("{{component.parent.env_dir}}/bin/python -m pip "
                  "install --upgrade pip")
+        if self.pip_version:
```

Comment:
I think this can be folded together with the pip install we're already running on line 22/23 ... at least I think it shouldn't be run twice ...